### ISSUE TITLE
Begin/Rescue/End block inside of a Lambda raises Style/RedundantBegin and formatter borks it

### ISSUE THREAD

**Issue by tadiou** [rubocop-hq/rubocop#6192]
## Expected behavior

```ruby
coerce_input ->(value, _ctx) do
  begin
    value.nil? ? nil : JSON.parse(value)
  rescue JSON::ParserError
    raise GraphQL::CoercionError, "cannot coerce `#{value.inspect}` to Hash value"
  end
end
```
is valid code (when stabby syntax is preferred in config).

## Actual behavior
After format it looks like this

```ruby
coerce_input ->(value, _ctx) do
  
    value.nil? ? nil : JSON.parse(value)
  rescue JSON::ParserError
    raise GraphQL::CoercionError, "cannot coerce `#{value.inspect}` to Hash value"
  
end 
```
which is invalid as heck. 

## Additional Info

Ruby's major GraphQL implementation often uses lambdas in arguments, we ended up sticking a begin/rescue/end block in one (because it works).

When it's separated by braces, a la

```ruby
coerce_method ->(value, _ctx) {
  begin
    value.nil? ? nil : JSON.parse(value)
  rescue JSON::ParserError
    raise GraphQL::CoercionError, "cannot coerce `#{value.inspect}` to Hash value"
  end
}
```

It works fine. Or if it was forced to be 

```ruby
coerce_input lambda(value, _ctx) do
  value.nil? ? nil : JSON.parse(value)
rescue JSON::ParseError
  raise GraphQL::CoercionError, "cannot coerce `#{value.inspect}` to Hash value"
end
```

Basically it tries to handle the stabby syntax in the same way that's allowed by the lambda syntax and ruby doesn't handle them the same way at all.

This exists in the same format whether it's being used as an argument or assigning the lambda to a variable.

## RuboCop version

`0.58.2 (using Parser 2.5.1.2, running on ruby 2.5.1 x86_64-linux)`
